_posts/2021-02-10-webhooks.md

```diff
@@ -0,0 +1,31 @@
+---
+title: "Webhooks in sensenet"
+author: [tusmester]
+image: "../img/posts/webhooks/header.png"
+tags: [webhooks, API, saas, event, service]
+redirect_to: https://www.sensenet.com/blog/2021-02-10-webhooks
+---
+
+Webhooks have (finally) arrived to sensenet!
+
+---
+
+In modern service platforms like sensenet developers need the possibility to do something when something happens in the service. In more technical terms: handling **events** is one of the most important things we need when building an enterprise application. 
+
+In the old days backend developers would handle this issue by developing a module that runs inside the application when a user uploads a file - but since everything has moved to the cloud, this is not an option. 
+
+Fortunately in a service-oriented environment we can do the same with [webhooks](https://en.wikipedia.org/wiki/Webhook). They are basically **HTTP requests** made by the service and a corresponding handler method on the other side. Developers can then do what their project requires in case of these events - for example archiving a document, starting a build in the cloud or sending a notification email.
+
+Many popular services offer webhooks. You can react to [code commits](https://docs.github.com/en/github/extending-github/about-webhooks) on GitHub or [email events](https://sendgrid.api-docs.io/v3.0/webhooks/retrieve-event-webhook-settings) in SendGrid. 
+
+<p align="center">
+<img src="/img/posts/webhooks/integrations.png" alt="integrations">
+</p>
+
+In sensenet webhooks are an easy way to **integrate our service with another**. They are usually activated by content changes. You can register a webhook for the usual content operations - for example when somebody uploads a file or changes an article.
+
+When such an event occurs, we call the URL you provided. It can be a custom application in your datacenter, a serverless function in the cloud or a 3rd party service like Slack or Netlify. We make these HTTP requests based on the **filter** you define: for example you are interested only in new file uploads in a certain library but not article changes. You also have some control of the request itself - for example you can add custom headers or url parameters required by your webhook handler.
```

Comment:
```suggestion
When such an event occurs, we call the URL you provided. It can be a custom application in your datacenter, a serverless function in the cloud or a 3rd party service like Slack or Netlify. We make these HTTP requests based on the **filter** you define: for example you are interested only in new file uploads in a certain library but not article changes. You also have some control over the request itself - for example you can add custom headers or url parameters required by your webhook handler.
```